examples/routing.js

```diff
@@ -0,0 +1,55 @@
+var plastiq = require('..');
+var h = plastiq.html;
+var router = require('../router');
+
+var objects = [
+  { name: 'one', id: 1 },
+  { name: 'two', id: 2 },
+  { name: 'three', id: 3 },
+  { name: 'four', id: 4 }
+];
+
+function render(model) {
+  return h('div',
+    h('.header', 'Hi ', model.name),
+    router(
+      router.page('/', function () {
+        return h('div',
+          h('h1', 'root'),
+          router.link('/objects', 'objects')
+        );
+      }),
+      router.page('/objects', function () {
+        return new Promise(function (result) {
+          setTimeout(function () {
+            result(objects);
+          }, 300);
+        });
+      }, function (objects) {
+        return h('div',
+          h('h1', 'objects'),
+          h('ul',
+            objects.map(function (o) {
+              return router.link('/objects/' + o.id, h('li', o.name));
```

Comment:
`router.link()` generates an anchor tag, so I guess you meant to nest the link inside the `h('li')`. 

I wonder if regular anchors (as in `h('a', ...)`) should automatically plug into the router. That seems like it would be a good reason for routing to be integrated into plastiq itself. The usual reason for using something like `router.link` instead of regular anchors is to plug into some "named routes" system, e.g.
```
router.link('showObject', { id = o.id }, o.name)
```

Personally I like named routes on bigger projects because: 1) the framework can throw an error when you link to bad URLs instead of when you follow those links and 2) your code is isolated from changes to the URL. 

However mandatory route names complicate route setup code. Perhaps a middle ground is that routes could have name == pattern by default. That would allow this:
```
router.link('/objects/:id', { id: o.id }, o.name)
```
...or even this:
```
router.link('/objects/:id', o.id, o.name)
```